src/presentation/test/integration/registration/generalPartner/address/choose-general-partner-principal-office-address.test.ts

```diff
@@ -0,0 +1,125 @@
+    appDevDependencies.cacheRepository.feedCache({
+      [appDevDependencies.transactionGateway.transactionId]: {
+        ["principal_office_address"]: {
+          postal_code: "ST6 3LJ",
+          premises: "",
+          address_line_1: "",
+          address_line_2: "",
+          locality: "",
+          country: ""
+        }
+      }
+    });
+  });
+
+  const URL = getUrl(CHOOSE_GENERAL_PARTNER_PRINCIPAL_OFFICE_ADDRESS_URL);
+  // const REDIRECT_URL = getUrl(CONFIRM_GENERAL_PARTNER_PRINCIPAL_OFFICE_ADDRESS_URL); TODO
+
+  describe("GET choose principal office address of the general partner page", () => {
+
+    it("should load the choose principal office address of the general partner page with Welsh text", async () => {
+      setLocalesEnabled(true);
+
+      const res = await request(app).get(URL + "?lang=cy");
+
+      expect(res.status).toBe(200);
+      testTranslations(res.text, cyTranslationText.address.chooseAddress.generalPartnerPrincipalOfficeAddress);
+    });
+
+    it("should load the choose principal office address of the general partner page with English text", async () => {
+      setLocalesEnabled(true);
+
+      const res = await request(app).get(URL + "?lang=en");
+
+      expect(res.status).toBe(200);
+      testTranslations(res.text, enTranslationText.address.chooseAddress.generalPartnerPrincipalOfficeAddress);
+    });
+
+    it("should populate the address list", async () => {
+      const res = await request(app).get(URL);
+
+      expect(res.status).toBe(200);
+      expect(res.text).toContain("2 Duncalf Street, Stoke-On-Trent, ST6 3LJ");
+      expect(res.text).toContain("The Lodge Duncalf Street, Castle Hill, Stoke-On-Trent, ST6 3LJ");
+      expect(res.text).toContain("4 Duncalf Street, Stoke-On-Trent, ST6 3LJ");
+      expect(res.text).toContain("6 Duncalf Street, Stoke-On-Trent, ST6 3LJ");
+    });
+
+    it("should return error page when gateway getListOfValidPostcodeAddresses throws an error", async () => {
+      appDevDependencies.addressLookUpGateway.setError(true);
+
+      const res = await request(app).get(URL);
+
+      expect(res.status).toBe(500);
+      expect(res.text).toContain(enTranslationText.errorPage.title);
+    });
+  });
+
+  describe("POST choose principal office address of the general partner page", () => {
+
+    it("should redirect to the next page and add select address to cache", async () => {
+      const res = await request(app)
+        .post(URL)
+        .send({
+          pageType: AddressPageType.chooseGeneralPartnerPrincipalOfficeAddress,
+          selected_address: `{
+          "postal_code": "ST6 3LJ",
+          "premises": "4",
+          "address_line_1": "DUNCALF STREET",
+          "address_line_2": "",
+          "locality": "STOKE-ON-TRENT",
+          "country": "GB-ENG"
+        }`
+        });
+
+      expect(res.status).toBe(302);
+      // expect(res.text).toContain(`Redirecting to ${REDIRECT_URL}`); TODO
```

Comment:
if these comments are TODOs, start them with // TODO, it's easier to find them and things like IDEs and Sonar will pick them up